dask/layers.py

```diff
@@ -408,25 +409,19 @@ def __init__(
         # much earlier.
         #
         # See https://github.com/dask/dask/pull/6051 for a detailed discussion.
-        annotations = annotations or {}
-        self._split_keys = None
-        if "priority" not in annotations:
-            annotations["priority"] = self._key_priority
-        super().__init__(annotations=annotations)
-
-    def _key_priority(self, key):
-        if key in self.get_split_keys():
-            return 1
-        return 0
+        self.annotations = self.annotations or {}
+        if "priority" not in self.annotations:
+            self.annotations["priority"] = {}
+        self.annotations["priority"]["__expanded_annotations__"] = None
+        self.annotations["priority"].update({_key: 1 for _key in self.get_split_keys()})
 
     def get_split_keys(self):
-        if self._split_keys is None:
-            self._split_keys = [
-                stringify((self.split_name, part_out, part_in))
-                for part_in in range(self.npartitions_input)
-                for part_out in self.parts_out
-            ]
-        return self._split_keys
+        # Return SimpleShuffleLayer "split" keys
+        return [
+            stringify((self.split_name, part_out, part_in))
+            for part_in in range(self.npartitions_input)
+            for part_out in self.parts_out
+        ]
```

Comment:
I suspect this is causing the regression. What I need for the refactoring is to drop the `__expanded_annotations__`